build_scripts/common/basic-ceph-server.sh

```diff
@@ -0,0 +1,209 @@
+#!/bin/sh
+
+# if any command fails, the script should exit
+set -e
+
+# enable some more output
+set -x
+
+# these variables need to be set
+[ -n "${GERRIT_HOST}" ]
+[ -n "${GERRIT_PROJECT}" ]
+[ -n "${GERRIT_REFSPEC}" ]
+
+# only use https for now
+GIT_REPO="https://${GERRIT_HOST}/${GERRIT_PROJECT}"
+
+# enable the Storage SIG Gluster and Ceph repositories
+dnf -y install centos-release-ceph epel-release
+
+BUILDREQUIRES="git bison cmake dbus-devel flex gcc-c++ krb5-devel libacl-devel libblkid-devel libcap-devel redhat-rpm-config rpm-build xfsprogs-devel lvm2"
+
+BUILDREQUIRES_EXTRA="libnsl2-devel libnfsidmap-devel libwbclient-devel userspace-rcu-devel"
+
+# basic packages to install
+case "${CENTOS_VERSION}" in
+    7)
+        yum install -y ${BUILDREQUIRES} ${BUILDREQUIRES_EXTRA} python2-devel
+    ;;
+    8s)
+        yum install -y ${BUILDREQUIRES}
+        yum install --enablerepo=powertools -y ${BUILDREQUIRES_EXTRA}
+        yum install -y libcephfs-devel
+    ;;
+    9s)
+       yum install -y ${BUILDREQUIRES}
+       yum install --enablerepo=crb -y ${BUILDREQUIRES_EXTRA}
+       yum install -y libcephfs-devel
+    ;;
+esac
+
+git clone --depth=1 ${GIT_REPO}
+cd $(basename "${GERRIT_PROJECT}")
+git fetch origin ${GERRIT_REFSPEC} && git checkout FETCH_HEAD
+
+# update libntirpc
+git submodule update --recursive --init || git submodule sync
+
+# cleanup old build dir
+[ -d build ] && rm -rf build
+
+mkdir build
+cd build
+
+( cmake ../src -DCMAKE_BUILD_TYPE=Maintainer -DUSE_FSAL_GLUSTER=OFF -DUSE_FSAL_CEPH=ON -DUSE_FSAL_RGW=OFF -DUSE_DBUS=ON -DUSE_ADMIN_TOOLS=ON && make) || touch FAILED
+make install
```

Comment:
I think we should have a single build script to avoid repititions for `cephfs`.

My thoughts is that we should have an option to build from repo or not. In case of weekly runs, I believe we should build in a single stage and then pass the binaries around. This way we can avoid the need to build for each and every test.

Since the weekly would involve executing against cephfs with other testing tools (pynfs, cthon04, etc..)